docs/querying/caching.md

```diff
@@ -22,63 +23,87 @@ title: "Query caching"
+
+## Where to enable caching
+
+**Per-segment cache** is available as follows:
+
+- On Historicals, the default. Enable segment-level cache population on Historicals for larger production clusters to prevent Brokers from having to merge all query results. When you enable cache population on Historicals instead of Brokers, the Historicals merge their own local results and put less strain on the Brokers.
+
+- On ingestion tasks in the Peon or Indexer service. Larger production clusters should enable segment-level cache population on task execution services only to prevent Brokers from having to merge all query results. When you enable cache population on task execution services instead of Brokers, the the task execution services to merge their own local results and put less strain on the Brokers.
+
+     Task executor services only support caches that store data locally. For example the `caffeine` cache. This restriction exists because the cache stores results at the level of intermediate partial segments generated by the ingestion tasks. These intermediate partial segments may not be identical across task replicas. Therefore task executor services ignore remote cache types such as `memcached`.
+
+- On Brokers for small production clusters with less than five servers. 
+
+     Do not use per-segment caches on the Broker for large production clusters. When `druid.broker.cache.populateCache` is `true` and query context parameter `populateCache` _is not_ `false`, Historicals return results on a per-segment basis without merging results locally thus negatively impacting cluster scalability.
+
+**Whole-query cache** is only available on Brokers.
+
+## Performance considerations for caching
+Caching enables increased concurrency on the same system, therefore leading to noticeable performance improvements for queries on Druid clusters handling throughput for concurrent, mixed workloads.
+
+If you are looking to improve response time for a single query or page load, you should ignore caching. In general, response time for a single task should meet performance objectives even when the cache is cold.
+
+During query processing, the per-segment cache intercepts the query and sends the results directly to the Broker. This way the query bypasses the data server processing threads. For queries requiring minimal processing in the Broker, cached queries are very quick. If work done on the Broker causes a query bottleneck, enabling caching results in little noticeable query improvement.
+
+The largest performance gains from segment caching tend to apply to `topN` and time series queries. The impact is less for `groupBy` queries. The same applies to queries with or without joins.
```

Comment:
> The impact is less for `groupBy` queries

I think this undersells it a little bit. There is still an impact here, it's just that in some cases the bottleneck may be in merging results on the broker. Not sure exactly how to wordsmith this, but if I read this as is it sounds like there isn't much of a gain for groupBy queries 
```suggestion
The largest performance gains from segment caching tend to apply to `topN` and time series queries. For `groupBy` queries, if the bottleneck is in the merging phase on the broker, the impact is less. The same applies to queries with or without joins.
```

> The same applies to ...

I don't think I understand this sentence

Reply:
```suggestion
The largest performance gains from segment caching tend to apply to `topN` and time series queries. The impact is less for `groupBy` queries.
```
I removed that sentence. I lifted it from a perf topic by @gianm and I can't find the source context now.